When building with `intel` or `oneapi` or ... as preferred compiler, use `gcc` for building `cmake`?

**Is your feature request related to a problem? Please describe.**

I am getting weird build failures for `cmake@3.27.9` with the latest `oneapi@2025.0.1`. Since `cmake` is simply a build tool that is not linked to any code we compile, I am wondering if we should build `cmake` with `gcc` when the preferred compiler is in this list:
- intel
- oneapi
- nvhpc

I would prefer to build `cmake` with the preferred compiler when the preferred compiler is any of:
- llvm
- apple-clang
- clang
- aocc

Am I missing a compiler? Comments? Suggestions?

**Describe the solution you'd like**

See above

**Additional context**

n/a

I was able to fix the bug, and it was not straightforward. If you ever see this when trying to build `cmake` with a compiler that clearly has C++11 support (which compiler doesn't in 2025):
```
CMake Error at CMakeLists.txt:93 (message):
  The C++ compiler does not support C++11 (e.g.  std::unique_ptr).


-- Configuring incomplete, errors occurred!
---------------------------------------------
Error when bootstrapping CMake:
Problem while running initial CMake
---------------------------------------------
```
Then check if your `spack` build stage directory is on an NFS mounted directory or not. If it is, change it to a local directory, a GPFS or Lustre mount, ... just not NFS. I will add something to the "Known issues" section about this.

The question remains, however. There is no reason to build `cmake` with Intel, especially because a `gcc` compiler must be present anyway as the backend. We already compile `gmake` with `gcc`. If we compile such basic build system utilities with `gcc` for the list of compilers suggested above, then we increase the possibility of reusing build caches across environments.

--- a/configs/common/packages.yaml
+++ b/configs/common/packages.yaml
@@ -40,8 +40,9 @@ packages:
       when: '%intel@:2020'
       message: '2.0.5 is the last version to use C++14, use with Intel Classic 2020 and earlier'
   cmake:
-    version: ['3.27.9']
-    require: '+ownlibs'
+    require:
+    - '@3.29:'
+    - '+ownlibs'
   # Attention - when updating also check the various jcsda-emc-bundles env packages
   crtm:
     require: '+fix'

--- a/configs/common/packages.yaml
+++ b/configs/common/packages.yaml
@@ -49,7 +49,9 @@ packages:
   ecbuild:
     require: '@3.7.2'
   eccodes:
-    require: '@2.33.0 +png +tools'
+    require:
+    - '@2.33.0'
+    - '+netcdf +png +tools'
   ecflow:
     require:
     - '@5.11.4 +ui'

--- a/configs/common/packages_intel.yaml
+++ b/configs/common/packages_intel.yaml
@@ -4,6 +4,9 @@ packages:
       blas: [intel-oneapi-mkl]
       fftw-api: [intel-oneapi-mkl]
       lapack: [intel-oneapi-mkl]
+  cmake:
+    require:
+    - '%gcc'
   ectrans:
     require:
     - '+mkl ~fftw'

--- a/configs/common/packages_oneapi.yaml
+++ b/configs/common/packages_oneapi.yaml
@@ -4,6 +4,9 @@ packages:
       blas: [intel-oneapi-mkl]
       fftw-api: [intel-oneapi-mkl]
       lapack: [intel-oneapi-mkl]
+  cmake:
+    require:
+    - '%gcc'
   ectrans:
     require:
     - '+mkl ~fftw'

--- a/spack-ext/repos/spack-stack/packages/base-env/package.py
+++ b/spack-ext/repos/spack-stack/packages/base-env/package.py
@@ -25,6 +25,7 @@ class BaseEnv(BundlePackage):
     depends_on("git", type="run")
     depends_on("wget", type="run")
     depends_on("curl", type="run")
+    depends_on("cloc", type="run")
 
     # I/O
     depends_on("zlib-api", type="run")

--- a/spack-ext/repos/spack-stack/packages/base-env/package.py
+++ b/spack-ext/repos/spack-stack/packages/base-env/package.py
@@ -41,5 +41,6 @@ class BaseEnv(BundlePackage):
     depends_on("py-wheel", type="run")
     depends_on("py-setuptools", type="run")
     depends_on("py-setuptools-scm", type="run")
+    depends_on("py-pytest", type="run")
 
     # There is no need for install() since there is no code.

